### ISSUE TITLE
[bug] Lazy validation fails with TypeError: __call__() takes 2 positional arguments but 3 were given

### ISSUE THREAD

**Issue by yahman72** [rochacbruno/dynaconf#662]
**Describe the bug**
Tried to introduce Lazy Validators as described in https://www.dynaconf.com/validation/#computed-values i.e.
```
from dynaconf.utils.parse_conf import empty, Lazy

Validator("FOO", default=Lazy(empty, formatter=my_function))
```
First bug (documentation): The above fails with
`   ImportError: cannot import name 'empty' from 'dynaconf.utils.parse_conf'`
--> "empty" seems to be now in `dynaconf.utils.functional.empty`
  
Second bug:
Lazy Validator fails with
`  TypeError: __call__() takes 2 positional arguments but 3 were given`

**To Reproduce**
```
from dynaconf.utils.parse_conf import Lazy
from dynaconf.utils.functional import empty

def lazy_foobar(s, v):
    return "foobar"

from dynaconf import Dynaconf
s = Dynaconf(
    validators=[Validator("FOOBAR", default=Lazy(empty, formatter=lazy_foobar)),],
)
print(s.FOOBAR)
```
stacktrace:
```
    print(s.FOOBAR)
  File "c:\ta\virtualenv\dconf\lib\site-packages\dynaconf\base.py", line 113, in __getattr__
    self._setup()
  File "c:\ta\virtualenv\dconf\lib\site-packages\dynaconf\base.py", line 164, in _setup
    settings_module=settings_module, **self._kwargs
  File "c:\ta\virtualenv\dconf\lib\site-packages\dynaconf\base.py", line 236, in __init__
    only=self._validate_only, exclude=self._validate_exclude
  File "c:\ta\virtualenv\dconf\lib\site-packages\dynaconf\validator.py", line 417, in validate
    validator.validate(self.settings, only=only, exclude=exclude)
  File "c:\ta\virtualenv\dconf\lib\site-packages\dynaconf\validator.py", line 198, in validate
    settings, settings.current_env, only=only, exclude=exclude
  File "c:\ta\virtualenv\dconf\lib\site-packages\dynaconf\validator.py", line 227, in _validate_items
    if callable(self.default)
TypeError: __call__() takes 2 positional arguments but 3 were given
```

**Expected behavior**
Work as documented in https://www.dynaconf.com/validation/#computed-values

**Environment (please complete the following information):**
 - OS: Windows 10 Pro
 - Dynaconf Version: 3.1.7 (also 3.1.4)
 - Frameworks in use: N/A

**Additional context**

**Comment by GibranHL0** [comment]
Hey, I found the issue. 👋

The problem behind this is within the `__call__` method as it states to receive the parameter `settings` to return a formatted tuple with the `self.value` and a dictionary with the `os.environ` and the `settings`. 

But within the `_validate_items` method, it invokes the `__call__` one, giving the values of `settings` and `self` (self is the `Validator` instance defined in the `default` argument while the `Settings` object is created). 

Then, the issue comes when the method is called and the `Validator` instance is not received.

But when checking the regular implementation, I found that the elements retrieved when the Validator is called are `settings` and the `Validator`. 

So, I was hesitating about which is going to be the best approach. 
- Do I change the `Lazy` `__call__` method to retrieve the values as the regular implementation (the `settings` and `Validator` objects)?
- Do I receive the `Validator` object and ignore it?
- Do I add a new if statement that checks whether is a regular implementation or a `Lazy` one to decide which elements should it send?
